lib/puppet/type/vm_vapp_property.rb

```diff
@@ -17,7 +17,11 @@
     desc 'The datacenter, vm name and visible label of the property split by a colon (:). Format dc1:vm1:label'
 
     munge do |value|
-      @resource[:datacenter], @resource[:vm_name], @resource[:label] = value.split(':',3)
+      if value =~ /^[a-zA-Z0-9\-\. ]+:[a-zA-Z0-9\-\. ]+:[a-zA-Z0-9\-\. ]+$/ then
```

Comment:
Note the original code doesn't restrict the datacenter and vm_name content (except that ':' is used as delimiter). 

Are you sure you don't want to allow '_'? 
```ruby
 /^[\w\-\. ]+:[\w\-\. ]:[\w\-\. ]$/
```